Mollier/SAM.Core.Mollier.UI/Modify/AddDivisionArea.cs

```diff
@@ -9,69 +9,112 @@ namespace SAM.Core.Mollier.UI
-                        else
-                        {
-                            label.Points.AddXY(Query.FindDivisionAreaCornerPoints(rh + deltaRelativeHumidity / 2, e + deltaEnthalpy / 2, "X", chartType, pressure), Query.FindDivisionAreaCornerPoints(rh + deltaRelativeHumidity / 2, e + deltaEnthalpy / 2, "Y", chartType, pressure));
-                        }
-                        label.Color = System.Drawing.Color.Transparent;
-                        label.Label = rectangles_points[index_1, index_2].Count.ToString();
-                        label.Tag = "GradientZoneLabel";
-
+                        result.Add(labelDivisionAreas(chart, pointsInAreas, mollierControlSettings, i, j));
                     }
                 }
             }
+            return result;
+        }  
+        private static Series labelDivisionAreas(Chart chart, List<MollierPoint>[,] pointsInAreas, MollierControlSettings mollierControlSettings, double relativeHumidity, double enthalpy )
+        {
+            ChartType chartType = mollierControlSettings.ChartType;
+            double pressure = mollierControlSettings.Pressure;
+            Tuple<int, int> areaIndex = getAreaIndex(relativeHumidity, enthalpy);
+
+            Series labelsSeries = chart.Series.Add("DivisionAreaLabels " + Guid.NewGuid().ToString());
+            labelsSeries.IsVisibleInLegend = false;
+            labelsSeries.ChartType = SeriesChartType.Point;
+            if (chartType == ChartType.Mollier)
+            {
+                labelsSeries.Points.AddXY(Query.FindDivisionAreaCornerPoints(relativeHumidity + deltaRelativeHumidity / 2, 
+                                                                            enthalpy + deltaEnthalpy / 2, "X", chartType, pressure), 
+                                          Query.FindDivisionAreaCornerPoints(relativeHumidity + deltaRelativeHumidity / 2, 
+                                                                            enthalpy + deltaEnthalpy / 2, "Y", chartType, pressure) - 0.5);
+            }
+            else
+            {
+                labelsSeries.Points.AddXY(Query.FindDivisionAreaCornerPoints(relativeHumidity + deltaRelativeHumidity / 2, 
+                                                                             enthalpy + deltaEnthalpy / 2, "X", chartType, pressure), 
+                                          Query.FindDivisionAreaCornerPoints(relativeHumidity + deltaRelativeHumidity / 2, 
+                                                                             enthalpy + deltaEnthalpy / 2, "Y", chartType, pressure));
+            }
+            labelsSeries.Color = System.Drawing.Color.Transparent;
+            labelsSeries.Label = pointsInAreas[areaIndex.Item1, areaIndex.Item2].Count.ToString();
+            labelsSeries.Tag = "GradientZoneLabel";
+            return labelsSeries;
+        }
+
+        private static Tuple<int, int> getAreaIndex(double relativeHumidity, double enthalpy)
```

Comment:
Get rid of private static Methods in Modify/Query. Use Func<> instead

            Func<double, double> func = new Func<double, double>((double value) => 
            {
		   double result = double.NaN;
		   //Do some stuff
                   return result;
            });
			
	    func.Invoke(10)